lib/GitUtils.jsx

```diff
@@ -0,0 +1,44 @@
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec);
+
+export default class GitUtils {
+
+    /**
+     * Takes in two git refs and returns a list of PR numbers of all PRs merged between those two refs
+     *
+     * @param {String} fromRef
+     * @param {String} toRef
+     * @returns {Promise}
+     */
+    getPullRequestsMergedBetween(fromRef, toRef) {
+        return new Promise((resolve, reject) => {
+            exec(`git log --format="%s" ${fromRef}...${toRef}`)
+                .then(({ stdout, stderr }) => {
+                    const commitMessages = stdout.split('\n');
+                    const pullRequestNumbers = commitMessages
```

Comment:
I think you could also use a global regex and `matchAll` to avoid looping and having a second function like this. I think this single function might achieve the same effect:

```javascript
getPullRequestsMergedBetween(fromRef, toRef) {
    return exec(`git log --format="%s" ${fromRef}...${toRef}`)
        .then(({ stdout, stderr }) => (
            [...stdout.matchAll(new RegExp('Merge pull request #(\\d{1,6})', 'g')]
                .map(match => match[1])
         ));
}
```

Also, you also don't necessarily need to wrap what you have in a `new Promise`.

Reply:
👍